src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -46,6 +46,8 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
             addr = daemons[0].ip if daemons[0].ip else self._inventory_get_fqdn(daemons[0].hostname)
             loki_host = build_url(scheme='http', host=addr, port=3100)
 
+            deps.append(daemons[0].name())
```

Comment:
the way `_calc_daemon_deps` works it adds all daemons of the matching type to the dependencies. This looks like it's only adding one loki daemon mno atter what. I'm guessing this would maybe work as we're meant to just have one anyway, but to match `_calc_daemon_deps` more precisely I feel like we should add all loki daemons here.

Reply:
Agree. I assume there was only one loki daemon. Because we don't support any HA stuff yet. I'll modify it to add dependencies for all loki daemons.